codec-memcache/src/main/java/io/netty/handler/codec/memcache/binary/AbstractBinaryMemcacheDecoder.java

```diff
@@ -104,8 +102,7 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
                         return;
                     }
 
-                    currentMessage.setKey(in.toString(in.readerIndex(), keyLength, CharsetUtil.UTF_8));
-                    in.skipBytes(keyLength);
+                    currentMessage.setKey(readBytes(ctx.alloc(), in, keyLength));
```

Comment:
we do it like this for the extras.. since keys and extras are handed off to someone else later, wouldn't the original buffer be freed at some point?